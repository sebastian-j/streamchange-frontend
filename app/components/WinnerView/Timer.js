import React from 'react';
import styled from 'styled-components';

const Span = styled.span`
  color: gray;
  margin-left: auto;
`;

export default class Timer extends React.Component {
  constructor(props) {
    super(props);
    this.state = { display: '', interval: null };
  }

  componentDidMount() {
    this.setState({ display: '0:00' });
    const start = new Date().getTime();
    const t = setInterval(() => {
      const now = new Date().getTime();
      const distance = now - start;
      const minutes = Math.floor((distance % (1000 * 60 * 60)) / (1000 * 60));
      const sec = Math.floor((distance % (1000 * 60)) / 1000);
      this.setState({ display: `${minutes}:${sec < 10 ? '0' : ''}${sec}` });
    }, 1000);
    this.setState({ interval: t });
  }

  componentWillUnmount() {
    clearInterval(this.state.interval);
  }

  render() {
    return <Span>{this.state.display}</Span>;
  }
}
