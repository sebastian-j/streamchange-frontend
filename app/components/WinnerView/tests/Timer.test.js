import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Timer from '../Timer';

const renderer = new ShallowRenderer();

describe('<Timer />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<Timer />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
