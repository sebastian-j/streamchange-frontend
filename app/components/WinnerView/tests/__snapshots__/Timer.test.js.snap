// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Timer /> should render and match the snapshot 1`] = `
<ForwardRef>
  
</ForwardRef>
`;
